core/eolearn/core/eodata_io.py

```diff
@@ -281,20 +303,69 @@ def _save(self, filesystem, data, path, file_format, compress_level=0):
             with gzip.GzipFile(fileobj=file_handle, compresslevel=compress_level, mode='wb') as gzip_file_handle:
                 self._write_to_file(data, gzip_file_handle, file_format)
 
-    @staticmethod
-    def _write_to_file(data, file, file_format):
+    def _write_to_file(self, data, file, file_format):
         """ Writes to a file
         """
         if file_format is FileFormat.NPY:
-            np.save(file, data)
-        elif file_format is FileFormat.PICKLE:
-            pickle.dump(data, file)
+            return np.save(file, data)
+
+        if file_format is FileFormat.GPKG:
+            return data.to_file(file, driver="GPKG", encoding="utf-8", layer=fs.path.basename(self.path))
```

Comment:
Do we want to allow the user to store several layers into the gpkg? 

Additionally, although `geopandas` doesn't require `driver="GPKG"` anymore, I'd keep it: explicit > implicit.

Reply:
No, it is not intended that we would have multiple layers. But it turns out that if you don't specify layer name then geopandas does the following:

- if `file` would be a filename then it uses its name for layer name,
- if `file` is a file object then it creates a random hash name for layer name which is not very nice.

So by setting `layer` parameter I'm reconstructing the `geopandas` functionality of nice names even though we are passing it a file object.

Yes, I also think it's better to be explicit by specifying `driver="GPKG"`.

Reply:
👍